src/memorylayout.jl

```diff
@@ -255,6 +283,70 @@ transposelayout(::ConjLayout{ML}) where ML = ConjLayout{typeof(transposelayout(M
 adjointlayout(::Type{T}, M::MemoryLayout) where T = transposelayout(conjlayout(T, M))
 
 
+# Layouts of PermutedDimsArrays
+"""
+    FirstMajor()
+
+is returned by `MemoryLayout(A)` for arrays of ndims(A) >= 3 for which `stride(A,1) == 1`,
+but unlike `ColumnMajor()` in that it does not demand that the other strides are monotonic.
+
+Dispatch on `FirstUnion = Union{FirstMajor, AbstractColumnMajor}` to ensure this property.
+
+Base.show_supertypes(ArrayLayouts.FirstMajor)
+"""
+FirstMajor
+
+const FirstUnion = Union{FirstMajor, AbstractColumnMajor}
```

Comment:
I don't think this needs to be in ArrayLayouts.jl: I tend to avoid union types due to compile time type inference issues (though something this simple won't be impacted to that) and its only used in one place.

If there's a strong reason to keep it, it needs a more descriptive name.

Reply:
OK, I removed this for now. 

The argument for it is that you would almost never want to dispatch on `UnitStride{1}` and exclude ColumnMajor, it just happens that there's no abstract type which fills this role.

Reply:
Hmm, I'm tempted to remove `IncreasingStrides` and define
```julia
const ColumnMajor = UnitStride{1}
```
though perhaps some more thought is needed...

Reply:
Do you mean `ColumnMajor = Union{DenseColumnMajor, UnitStride{1}}` as a replacement for `AbstractColumnMajor`?

One could also re-organise the type tree so that these two share a supertype. Does anything care about increasing strides, as opposed to just `DenseColumnMajor` or `UnitStride{1}`? 

However I also need `UnitStride{2}`, and clearly the tree can't depend on `D`.